src/httpserver.cpp

```diff
@@ -601,8 +612,21 @@ void HTTPRequest::WriteReply(int nStatus, const std::string& strReply)
     struct evbuffer* evb = evhttp_request_get_output_buffer(req);
     assert(evb);
     evbuffer_add(evb, strReply.data(), strReply.size());
-    HTTPEvent* ev = new HTTPEvent(eventBase, true,
-        std::bind(evhttp_send_reply, req, nStatus, (const char*)nullptr, (struct evbuffer *)nullptr));
+    auto req_copy = req;
```

Comment:
Why do we now need to copy req where we weren't previously?

Reply:
std::bind copied in req by value, but passing "this" into the lambda and using req would use this->req in the callback, which would blow up because req is set to nullptr a few lines below.

This is clunky, but I'm not sure how else to pass a class member into a lambda by value in c++11.

Reply:
If the lambda were changed to simply say "req" (not "req&") then it is copied by value (in this case the pointer is copied).

Reply:
Class members need to be captured with 'this', so unfortunately that doesn't work.